src/runtime/virtcontainers/nydusd.go

```diff
@@ -0,0 +1,424 @@
+			dialer := &net.Dialer{
+				Timeout:   dialTimout,
+				KeepAlive: keepAlive,
+			}
+			return dialer.DialContext(ctx, "unix", sock)
+		},
+	}, nil
+}
+
+func (c *NydusClient) CheckStatus() (DaemonInfo, error) {
+	resp, err := c.httpClient.Get(fmt.Sprintf("http://unix%s", infoEndpoint))
+	if err != nil {
+		return DaemonInfo{}, err
+	}
+	defer resp.Body.Close()
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return DaemonInfo{}, err
+	}
+	var info DaemonInfo
+	if err = json.Unmarshal(b, &info); err != nil {
+		return DaemonInfo{}, err
+	}
+	return info, nil
+}
+
+func (c *NydusClient) Mount(mountPoint string, mr *MountRequest) error {
+	requestURL := fmt.Sprintf("http://unix%s?mountpoint=%s", mountEndpoint, mountPoint)
+	body, err := json.Marshal(mr)
+	if err != nil {
+		return errors.Wrap(err, "failed to create mount request")
+	}
+
+	resp, err := c.httpClient.Post(requestURL, contentType, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNoContent {
+		return nil
+	}
+	return handleMountError(resp.Body)
+}
+
+func (c *NydusClient) Umount(sharedMountPoint string) error {
+	requestURL := fmt.Sprintf("http://unix%s?mountpoint=%s", mountEndpoint, sharedMountPoint)
+	req, err := http.NewRequest(http.MethodDelete, requestURL, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNoContent {
+		return nil
+	}
+	return handleMountError(resp.Body)
+}
+
+func handleMountError(r io.Reader) error {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	var errMessage ErrorMessage
+	if err = json.Unmarshal(b, &errMessage); err != nil {
+		return err
+	}
+	return errors.New(errMessage.Message)
+}
+
+/*
+   rootfs mount format: Type: fuse.nydus-overlayfs, Source: overlay
+   Options：[lowerdir=/foo/lower2:/foo/lower1,upperdir=/foo/upper,workdir=/foo/work,extraoption={source: xxx, config: xxx, snapshotdir: xxx}]
+*/
+
+type extraOption struct {
+	Source      string `json:"source"`
+	Config      string `json:"config"`
+	Snapshotdir string `json:"snapshotdir"`
+}
+
+const extraOptionKey = "extraoption="
+
+func parseExtraOption(options []string) (*extraOption, error) {
+	extraOpt := ""
+	for _, opt := range options {
+		if strings.HasPrefix(opt, extraOptionKey) {
+			extraOpt = strings.TrimPrefix(opt, extraOptionKey)
+		}
+	}
+	if len(extraOpt) == 0 {
+		return nil, errors.New("no extraoption found")
+	}
+	opt, err := base64.StdEncoding.DecodeString(extraOpt)
+	if err != nil {
+		return nil, errors.Wrap(err, "base64 decoding err")
+	}
+	no := &extraOption{}
+	if err := json.Unmarshal(opt, no); err != nil {
+		return nil, errors.Wrapf(err, "json unmarshal err")
+	}
+	if len(no.Config) == 0 || len(no.Snapshotdir) == 0 || len(no.Source) == 0 {
+		return nil, fmt.Errorf("extra option is not correct, %+v", no)
+	}
+	return no, nil
```

Comment:
This is big "busy": could you space out the code with a single blank line between chunks to make it more readable?

Reply:
done